File: frb_example/pure_dart/lib/src/rust/api/structure.dart
// ignore_for_file: invalid_use_of_internal_member

import '../frb_generated.dart';
import 'package:flutter_rust_bridge/flutter_rust_bridge_for_generated.dart';

Future<StructWithOneFieldTwinNormal> funcStructWithOneFieldTwinNormal(
        {required StructWithOneFieldTwinNormal arg, dynamic hint}) =>
    RustLib.instance.api.funcStructWithOneFieldTwinNormal(arg: arg, hint: hint);

Future<StructWithTwoFieldTwinNormal> funcStructWithTwoFieldTwinNormal(
        {required StructWithTwoFieldTwinNormal arg, dynamic hint}) =>
    RustLib.instance.api.funcStructWithTwoFieldTwinNormal(arg: arg, hint: hint);

Future<StructWithZeroFieldTwinNormal> funcStructWithZeroFieldTwinNormal(
        {required StructWithZeroFieldTwinNormal arg, dynamic hint}) =>
    RustLib.instance.api
        .funcStructWithZeroFieldTwinNormal(arg: arg, hint: hint);

Future<TupleStructWithOneFieldTwinNormal> funcTupleStructWithOneFieldTwinNormal(
        {required TupleStructWithOneFieldTwinNormal arg, dynamic hint}) =>
    RustLib.instance.api
        .funcTupleStructWithOneFieldTwinNormal(arg: arg, hint: hint);

Future<TupleStructWithTwoFieldTwinNormal> funcTupleStructWithTwoFieldTwinNormal(
        {required TupleStructWithTwoFieldTwinNormal arg, dynamic hint}) =>
    RustLib.instance.api
        .funcTupleStructWithTwoFieldTwinNormal(arg: arg, hint: hint);

class StructWithOneFieldTwinNormal {
  final int a;

  const StructWithOneFieldTwinNormal({
    required this.a,
  });
}

class StructWithTwoFieldTwinNormal {
  final int a;
  final int b;

  const StructWithTwoFieldTwinNormal({
    required this.a,
    required this.b,
  });
}

class StructWithZeroFieldTwinNormal {
  const StructWithZeroFieldTwinNormal();
}

class TupleStructWithOneFieldTwinNormal {
  final int field0;

  const TupleStructWithOneFieldTwinNormal({
    required this.field0,
  });
}

class TupleStructWithTwoFieldTwinNormal {
  final int field0;
  final int field1;

  const TupleStructWithTwoFieldTwinNormal({
    required this.field0,
    required this.field1,
  });
}

File: frb_example/pure_dart/rust/src/api/structure.rs
use flutter_rust_bridge::frb;

pub struct StructWithZeroFieldTwinNormal {}

pub fn func_struct_with_zero_field_twin_normal(
    arg: StructWithZeroFieldTwinNormal,
) -> StructWithZeroFieldTwinNormal {
    arg
}

pub struct StructWithOneFieldTwinNormal {
    pub a: i32,
}

pub fn func_struct_with_one_field_twin_normal(
    arg: StructWithOneFieldTwinNormal,
) -> StructWithOneFieldTwinNormal {
    arg
}

pub struct StructWithTwoFieldTwinNormal {
    pub a: i32,
    pub b: i32,
}

pub fn func_struct_with_two_field_twin_normal(
    arg: StructWithTwoFieldTwinNormal,
) -> StructWithTwoFieldTwinNormal {
    arg
}

pub struct TupleStructWithOneFieldTwinNormal(pub i32);

pub fn func_tuple_struct_with_one_field_twin_normal(
    arg: TupleStructWithOneFieldTwinNormal,
) -> TupleStructWithOneFieldTwinNormal {
    arg
}

pub struct TupleStructWithTwoFieldTwinNormal(pub i32, pub i32);

pub fn func_tuple_struct_with_two_field_twin_normal(
    arg: TupleStructWithTwoFieldTwinNormal,
) -> TupleStructWithTwoFieldTwinNormal {
    arg
}

#[frb]
pub struct StructWithFieldRenameTwinNormal {
    #[frb(name = "renamed_field")]
    pub class: i32,
}

pub fn func_for_struct_with_field_rename_twin_normal(
    arg: StructWithFieldRenameTwinNormal,
) -> StructWithFieldRenameTwinNormal {
    arg
}

pub struct StructWithDartKeywordFieldTwinNormal {
    pub class: i32,
    pub interface: i64,
}

pub fn func_for_struct_with_dart_keyword_field_twin_normal(
    arg: StructWithDartKeywordFieldTwinNormal,
) -> StructWithDartKeywordFieldTwinNormal {
    arg
}
